packages/pigeon/lib/cpp_generator.dart

```diff
@@ -131,7 +131,7 @@ void _writeCodecSource(Indent indent, Api api, Root root) {
         indent.scoped('{', '}', () {
           indent.writeln('stream->WriteByte(${customClass.enumeration});');
           indent.writeln(
-              'WriteValue(std::any_cast<${customClass.name}>(*custom_value).ToEncodableMap(), stream);');
+              'WriteValue(std::any_cast<${customClass.name}>(*custom_value).ToEncodableList(), stream);');
```

Comment:
This is another one that was already wrong that we should fix; `WriteValue` takes an EncodableValue, so the first argument needs to be wrapped.

(Alternately `ToEncodableList` could be changed to `ToEncodableValue` and return an `EncodableValue`, doing the wrapping internally; either way works for me. The latter would require some more unit test adjustments.)